server/svix-server/src/core/operational_webhooks.rs

```diff
@@ -0,0 +1,181 @@
+// SPDX-FileCopyrightText: © 2022 Svix Authors
+// SPDX-License-Identifier: MIT
+
+//! Module defining an interface for sending webhook events about the service.
+
+use std::sync::Arc;
+
+use chrono::{DateTime, Utc};
+use http::StatusCode;
+use serde::Serialize;
+use svix::api::{MessageIn, Svix, SvixOptions};
+
+use super::{
+    security::{generate_management_token, Keys},
+    types::{
+        ApplicationId, ApplicationUid, EndpointId, EndpointUid, MessageAttemptId, MessageId,
+        MessageUid, OrganizationId,
+    },
+};
+use crate::{
+    db::models::messageattempt,
+    error::{HttpError, Result},
+};
+
+/// Sent when an endpoint has been automatically disabled after continuous failures.
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EndpointDisabledEvent<'a> {
+    pub app_id: &'a ApplicationId,
+    pub app_uid: Option<&'a ApplicationUid>,
+    pub endpoint_id: &'a EndpointId,
+    pub endpoint_uid: Option<&'a EndpointUid>,
+    pub fail_since: DateTime<Utc>,
+}
+
+/// Sent when an endpoint is created, updated, or deleted
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EndpointEvent<'a> {
```

Comment:
Should this structure include a field for event_type?

Reply:
No, it's automatically done in the enum (search this file for `endpoint.updated` for example).
The enum automatically puts this (the data) into `data`) and the name into `type`.